client/vscode/src/chatActivation.ts

```diff
@@ -21,6 +24,30 @@ export function registerChat(languageClient: LanguageClient, extensionUri: Uri)
         languageClient.info(`vscode client: Received ${JSON.stringify(message)} from chat`)
 
         switch (message.command) {
+            case chatRequestType.method:
+                const partialResultToken = uuidv4()
+
+                languageClient.onProgress(chatRequestType, partialResultToken, partialResult => {
+                    if (partialResult.body) {
+                        panel.webview.postMessage({
+                            command: 'aws/chat/sendChatPrompt',
+                            params: partialResult,
+                            isPartialResult: true,
+                            tabId: message.params.tabId,
+                        })
+                    }
+                })
+
+                languageClient
+                    .sendRequest(chatRequestType, Object.assign(message.params, { partialResultToken }))
+                    .then((chatResult: ChatResult) => {
+                        panel.webview.postMessage({
+                            command: 'aws/chat/sendChatPrompt',
+                            params: chatResult,
+                            tabId: message.params.tabId,
```

Comment:
As discussed, I suggest to move `tabId` into `params`, so that copying `tabId` should not be responsibility of all the clients. Can be handled separately.